### ISSUE TITLE
Convert Rythmos tests to Tempus 

### ISSUE THREAD

**Issue by ikalash** [SNLComputation/Albany#583]
To remove dependency on Rythmos.

**Comment by ikalash** [comment]
I have pretty much finished this.  There is only one issue, which I mentioned at the Albany meeting.  For the NSRayleighBernard2D test case, tempus and rythmos compute slightly different results. The differences start off immediately.  @gahansen , could you please have a look at this, as we discussed at the meeting?  I'm ok with rebaselining the test, but I'd like for someone to look at the differences and to try to understand them.  The relevant input files are:

https://github.com/SNLComputation/Albany/blob/master/tests/small/NSRayleighBernard2D/input.yaml
https://github.com/SNLComputation/Albany/blob/master/tests/small/NSRayleighBernard2D/inputTempus.yaml

Thanks!  

If you run them, you should be able to see the differences.

**Comment by gahansen** [comment]
I took a good look at the differences between the Rythmos and Tempus versions of this test. I feel that it is perfectly fine to re-baseline this test and proceed with only the Tempus version. I conclude that, due to implementation differences between Tempus and Rythmos - specifically the initialization of each - that we should expect the different behavior that we are seeing.

The physics behind this test is the 2D Rayleigh-Bernard convection cell problem. The formation of the convection cells will be sensitive to the initial condition that starts the transient; indeed the initial setup of Rythmos and Tempus are quite different in the first three time steps shown in these paraview images:

Rythmos saves a time = 0 solution equal to one in the Exodus file:
![t_zero](https://user-images.githubusercontent.com/5393804/81990225-f8b28380-95fb-11ea-8bdd-ee416002f0de.png)
Tempus backward Euler saves its initial solution at time = 1, equal to zero in the Exodus file. Tempus BDF2 saves a time = 1 solution of negative one:
![t_one](https://user-images.githubusercontent.com/5393804/81990270-154ebb80-95fc-11ea-8508-d578b7d4e0a3.png)
At time = 2, all three have settled on a solution near zero, but one sees that slight differences in the solution exist due to the different initialization characteristics of the three approaches:
![t_three](https://user-images.githubusercontent.com/5393804/81990303-24ce0480-95fc-11ea-976e-78f090668903.png)

Here is a movie of the transient that now develops due to these different initialization characteristics:

![output](https://user-images.githubusercontent.com/5393804/81991043-99ee0980-95fd-11ea-880b-8fe52c8c0337.gif)

I plotted these data using a vertical line through the center of the Bernard cells, the green curves show three solutions of the (solution_1) field; Rythmos backward Euler, Tempus backward Euler, and Tempus BDF2. The two Tempus solutions are similar, however the differences in the time integrator cause the convection cells to evolve at different rates. The Rythmos result is actually "flipped" - due to the difference in the initial state the cells evolve opposite of the direction that they do in the Tempus result. 

Such behavior would be expected in an unstable problem like this one - it will be pretty sensitive to the initial condition.

To summarize - I feel it is fine to re-baseline this problem and move forward without concern...

**Comment by bartgol** [comment]
Thanks for the in-depth analysis Glen! I'll let Irina close since she's owning the time integration stuff.

**Comment by ikalash** [comment]
Yes, thanks for looking at this, Glen!  I'll go ahead and rebaseline, as well as remove the Rythmos dependency.  

I am a little bit confused as to why Tempus saves the initial solution at time 1, unlike Rythmos.  Is that a bug in your view?
